### ISSUE TITLE
Support membership to multiple teams

### ISSUE THREAD

**Issue by mpetito** [a85/POSTMan-Chrome-Extension#7150]
**Is your feature request related to a problem? Please describe.**

I work on multiple projects for various clients who each want to have ownership over their own “team” and collections.  These clients will also have team members and so naturally they'd like to sign up for their own team subscription and invite me to join their team.

Unfortunately, Postman accounts are limited to single team membership.  This means that I would need a unique email to establish separate accounts each with their own passwords for every team I want to join.

**Describe the solution you'd like**

I would like to be able to join multiple teams with a single email address / account.  Switching teams could be handled like switching accounts in the client.

**Describe alternatives you've considered**

 - Using a single shared login per client instead of using the team functionality.
 - Using tools other than Postman.

**Additional context**

Related issues:

Unable to join multiple teams #6403
> One of my team member is currently in another team and they're unable to join my team.

Allow Developers with a pro license to be added to a customer's pro workspace at no extra cost. #5530
> What ends up happening is that my customer shares their login with me (bad for security)...

**Comment by wywarren** [comment]
I just emailed the Postman team about the exact same issue. They pointed me to this thread, and also suggested using a single enterprise license and creating private workspaces for clients. The downside is you’d have to be in charge of billing your clients as their accounts would be managed under your license. For most higher end clients, they require ownership and management of their own APIs so I’m stuck in the same boat right now.

**Comment by Carl-Hugo** [comment]
Having the possibility for Pro accounts to join multiple teams would be great, indeed, especially joining different organizations. Implementing this could be as simple as _securing workspaces_ (so less impact on billing and teams code; I guess):

- One user (team) owns a Workspace
- Other team members can join (based on whatever policies are configured using the existing features)
- External Pro account could be given access to one or more Workspaces (read or write)

**Comment by LukasV** [comment]
This is big issue for us. I've got a PRO license myself and I convinced my client to buy one as well. Only to realize we can't actually share a workspace? I know I should have red the conditions better but still this is pretty ridiculous. How do other freelancers (working for multiple clients) deal with the situation? Probably with some other tools, right? Shame. :(

**Comment by harrisrobin** [comment]
🤦‍♂️

**Comment by harrisrobin** [comment]
Surprised this is not possible.

this gets a +1 from me as well!

**Comment by matejskubic** [comment]
Please move this issue upper in priority line as this feature is crucial to work wit multiple teams without complication generating multiple user accounts.

**Comment by mollymc1** [comment]
We experienced this when we changed from a free account to a paid account (new team). Postman should at least include a pop up window letting people know this when they go to join another team (reminder to save data would also be helpful). It would save a lot of headaches. It wiped out all of our prior team work and we had to replicate it (waste of time).

**Comment by kckolz** [comment]
+1 I currently manage 3 teams all with separate postman collections. This is a very large pain point for me.

**Comment by EvilJordan** [comment]
Two years later, I asked the same question and was pointed to this thread. Come on, guys!

**Comment by Shujee** [comment]
Stumbled upon this limitation today and was surprised just like everyone else. I'm a dev and need to work with many clients/teams in parallel. Having to create multiple Postman accounts shouldn't be required for this kind of stuff.

**Comment by brendan-zeus** [comment]
This is a large pain point. Please fix

**Comment by jtchen2k** [comment]
+1, two years later and I'm still looking for solutions.

**Comment by melvinotieno** [comment]
Will this even be supported down the line?

**Comment by ArjunSingh-PM** [comment]
Hi @melvinotieno! Yes, we will be working towards supporting the ability to join multiple teams using a single user account by mid-2022.

**Comment by joshorr** [comment]
I'm in the same situation, just discovered this limitation.

**Comment by marcusjroberts** [comment]
This one is becoming a big issue with us. My engineering team collaborates on workspaces between themselves, but we also need to be able to share specific workspaces with a third-party API developer. We all have paid accounts, just on separate teams, and we're unable to collaborate on a single workspace. Possibly my use case is slightly different but I didn't want to open a new issue, despite the lack of traction on this one. Ideally we should be able to pick individual workspaces to make available to different teams.

The alternative is that we create and pay for a set of additional accounts for the third-party, but potentially on those engineer's private email addresses as their work email addresses are already in use. Or we go public on the workspace but that's also not feasible.

Are there any details about what's being implemented for mid-2022? Will it just be that my existing work email address will be usable on the third-party's team, but therefore still require a subscription on that team, etc? Or once an email address is a paid-for account, it's treated as paid-for regardless of the team(s) it's in?

**Comment by pankajkumar005** [comment]
@marcusjroberts  As @ArjunSingh-PM mentioned in the previous message, we are working on supporting single user to be part of more than one team. That may solve your use case up to some extent.
As far as "specific workspace being available to different teams" is concerned, we have this in our roadmap and will most likely be available towards the end of the year.

**Comment by Julian88Tex** [comment]
@pankajkumar005 do you suspect resolving this issues would also implement #10845?

**Comment by wompeter** [comment]
Is this still on track for mid-2022?

**Comment by pankajkumar005** [comment]
@wompeter Yes. The team is actively working on this. We hope to start the slow rollout to the users, by end of June or the beginning of July.

**Comment by kurtosys-ronellegriffiths** [comment]
This is so stupid, why can you only belong to one team? Pls add this functionality

**Comment by pankajkumar005** [comment]
@ronelle-griffiths  "Ability to join multi-team" is in the final phase of testing right now. We will start rolling this out in a couple of weeks. Please bear with us till then :)

**Comment by earthpyy** [comment]
Looking forward to this feature to be released!

**Comment by cumulus-william** [comment]
Is there an ETA on this feature being rolled out? I need to join a new team so if this feature is a few days away I'll wait.

**Comment by ArjunSingh-PM** [comment]
@cumulus-william Could you please raise a request [here](https://support.postman.com/hc/en-us/requests/new) or reach out to us directly at help@postman.com with your account details? We will be happy to enable this feature for you at the earliest.

**Comment by jalanubha** [comment]
Hello Postman Team, Is there an update on this? Really need to join multiple teams.

**Comment by ArjunSingh-PM** [comment]
Hi @jalanubha, thanks for reaching out. Could you please raise a request [here](https://support.postman.com/hc/en-us/requests/new) or reach out to us directly at help@postman.com with your account details?

**Comment by khairul-cprvision** [comment]
It seems like Postman is not targeted at large enterprises that collaborate across multiple entities

**Comment by The-Anomaly** [comment]
Hi Postman Team. Is there any update on this?

**Comment by Scrip7** [comment]
The way they rolled out this feature in production is mind-blowing.
I was part of an "empty" Postman team I created half a year ago.
By empty, I mean it had no members or collection or anything. It was a barebone team with no data whatsoever.

So I decided to leave and delete the team because I had all my collection on my "Personal Workspace."
And I did it.

It turned out my action deleted my entire account's data without warning!
Postman terminated my session, and I had to re-register on their website
Once I logged into my account, it was empty.
Nothing was there, even my custom profile photo I uploaded long ago.

In simpler words, their back-end deleted my entire account because I left an empty team, and this is how stupid the scenario is.

So I contacted their support; here's the response I got:

> Sorry that you have lost access to your workspace and the collections inside it after leaving the team.
> As per the new workflow after the roll-out of[ multi-team feature ](https://github.com/postmanlabs/postman-app-support/issues/7150), the "Personal Workspace" now becomes a team-owned workspace, and if the user leaves a team, it means that you will lose access to both personal and group workspace.

I understand the need for such a feature where users can share their data with multiple teams,
but having such a flaw where we can lose our entire account by leaving an empty team is unacceptable.

Please investigate and fix this issue.
Thank you so much for your attention and participation.

**Comment by matmanna** [comment]
> but having such a flaw where we can lose our entire account by leaving an empty team is unacceptable.

+1 - That sounds like a really messed up backend system. Now I'm too scared to even try leaving a team.

**Comment by emileswain** [comment]
Are there any links that explain how to create a new team when i'm already a member of a team. This is rediculous.

**Comment by kcaashish** [comment]
Is there anyway we can keep our personal workspace separate from any team? My personal workspace should belong to me and not be part of a team, and that can be true for the creator of the team as well. Ability to separate a personal workspace which became a part of the team, would also help. @postman-support

**Comment by ArjunSingh-PM** [comment]
You can join multiple teams in Postman using a single account and seamlessly switch between your teams. 

Refer: https://postman.slack.com/archives/C091X8PBQN7/p1752274276363829

**Comment by VannaDii** [comment]
> You can join multiple teams in Postman using a single account and seamlessly switch between your teams.
> 
> Refer: https://postman.slack.com/archives/C091X8PBQN7/p1752274276363829

@ArjunSingh-PM, how is this a useful response when it requires authentication that the public doesn't have?!

Can you please provide a legitimate response with actual information that folks can actually access?
